src/odemis/driver/test/ephemeron_test.py

```diff
@@ -0,0 +1,253 @@
+        cls.acquired_data: Optional[model.DataArray] = None
+        cls.dwell_time_values = [cls.ebic_det.dwellTime.range[0], 6e-6, 10e-6, 11.9e-6, 12e-6, 50e-6, 1.995e-3, 1.996e-3]
+        cls.resolution_values = [(140, 100), (300, 420), (2100, 2000)]
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.ebic_det.terminate()
+
+    def test_dwell_time(self):
+        """
+        Test for setting the dwell time of the detector.
+        """
+        print(self.ebic_det.dwellTime.range)
+        for dt in self.dwell_time_values:
+            self.ebic_det.dwellTime.value = dt
+            self.assertLessEqual(self.ebic_det.dwellTime.value, dt)
+
+    def test_acquisition(self):
+        """
+        Test acquisitions with various dwell times and resolutions.
+        """
+        # TODO: for now, this assumes the hardware pixel trigger is either sent separately, or always
+        # active. (For the simulator, it simulates as if is always active)
+
+        for i, dt in enumerate(self.dwell_time_values):
+            # set the resolution of the region of acquisition to 140 x 100, force it non-squared
+            if dt < 1e-3:
+                res = (140, 500 + i)
+            else:
+                # Don't make it too long for the long dwell time
+                res = (100 + i, 100)
+            self.ebic_det.resolution.value = res
+            self.ebic_det.dwellTime.value = dt
+
+            act_dt = self.ebic_det.dwellTime.value
+            logging.debug("Expected duration for %s px @ %s s: %s s", res, act_dt,
+                          res[0] * res[1] * act_dt)
+
+            start_time = time.time()
+            da = self.ebic_det.data.get()
+
+            # check if the data is in the right shape
+            self.assertEqual(da.shape[::-1], self.ebic_det.resolution.value)
+            self.assertEqual(da.dtype.type, numpy.float64)
+            self.assertGreater(da.metadata[model.MD_ACQ_DATE], start_time)  # Should be a tiny bit later
+
+    def test_acquisition_stop(self):
+        # set the resolution of the region of acquisition to 500 x 480, force it non-squared
+        self.ebic_det.resolution.value = (500, 480)
+        self.ebic_det.dwellTime.value = 100e-6
+        # => ~20s
+
+        self.acquired_data = None
+        self.ebic_det.data.subscribe(self.receive_ebic_data)
+
+        # stop the acquisition after a few seconds
+        time.sleep(4)
+        self.ebic_det.data.unsubscribe(self.receive_ebic_data)
+        self.assertIsNone(self.acquired_data)  # there should be no data acquired
+
+        # check if the state is changed to stopped
+        time.sleep(1)
+        self.assertEqual(self.ebic_det._opc_client.controller_state, ephemeron.STATE_NAME_IDLE)
+
+    def test_request_very_high_resolution(self):
+        """
+        This test should fire a timeout exception in the acquisition thread which is handled.
+        Take the res (4096, 3072) of the e-beam as resolution to scan.
```

Comment:
from this description it reads like you want to test if a timeout exception is fired, or at least that the acquisition stops before the very large acquisition is done. But the code looks very different. What is the purpose of this test?

Reply:
This description was outdated, I've fixed it.